Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Specific/DownsampleSinglePassMipChainPass.cpp

```diff
@@ -0,0 +1,297 @@
+            m_indicesAreInitialized = true;
+        }
+
+        void DownsampleSinglePassMipChainPass::GetInputInfo()
+        {
+            // Get the input/output mip chain attachment for this pass (at binding 0)
+            AZ_Assert(
+                GetInputOutputCount() > 0, "[DownsampleSinglePassMipChainPass '%s']: must have an input/output", GetPathName().GetCStr());
+            PassAttachment* attachment = GetInputOutputBinding(0).GetAttachment().get();
+
+            if (attachment)
+            {
+                m_mipLevelCount = attachment->m_descriptor.m_image.m_mipLevels;
+                m_inputImageSize[0] = attachment->m_descriptor.m_image.m_size.m_width;
+                m_inputImageSize[1] = attachment->m_descriptor.m_image.m_size.m_height;
+            }
+        }
+
+        void DownsampleSinglePassMipChainPass::CalculateBaseSpdImageSize()
+        {
+            const auto adjust = [&] (uint32_t value) -> uint32_t
+            {
+                const auto logValue = static_cast<uint32_t>(floorf(log2f(value * 1.f)));
+                if (static_cast<uint32_t>(1 << logValue) == value)
+                {
+                    return value;
+                }
+                else
+                {
+                    return static_cast<uint32_t>(1 << (logValue + 1));
+                }
+            };
+            m_baseSpdImageSize[0] = adjust(m_inputImageSize[0]);
+            m_baseSpdImageSize[1] = adjust(m_inputImageSize[1]);
+        }
+
+        void DownsampleSinglePassMipChainPass::BuildPassAttachment()
+        {
+            // Build "Mip6" image attachment.
+            {
+                const auto width = GetMax<uint32_t>(1, m_baseSpdImageSize[0] >> GloballyCoherentMipIndex);
+                const auto height = GetMax<uint32_t>(1, m_baseSpdImageSize[1] >> GloballyCoherentMipIndex);
+                m_mip6ImageDescriptor =
+                    RHI::ImageDescriptor::Create2D(
+                        RHI::ImageBindFlags::ShaderReadWrite,
+                        width, 
+                        height,
+                        RHI::Format::R32G32B32A32_FLOAT);
+
+                m_mip6PassAttachment = aznew PassAttachment();
+                m_mip6PassAttachment->m_name = "Mip6";
+                const Name attachmentPath(AZStd::string::format(
+                    "%s.%s",
+                    GetPathName().GetCStr(),
+                    m_mip6PassAttachment->m_name.GetCStr()));
+                m_mip6PassAttachment->m_path = attachmentPath;
+                m_mip6PassAttachment->m_lifetime = RHI::AttachmentLifetimeType::Transient;
+                m_mip6PassAttachment->m_descriptor = m_mip6ImageDescriptor;
+                m_ownedAttachments.push_back(m_mip6PassAttachment);
+
+                RHI::AttachmentLoadStoreAction action;
+                // Components: (min, average, max, weight)
+                action.m_clearValue = RHI::ClearValue::CreateVector4Float(FLT_MAX, 0.f, 0.f, 0.f);
```

Comment:
Do we need to clear the view? I would assume that the data is fully written to each frame

Reply:
A thread group that the threads do early return does not write its data.  As an option, we can write it just before the early return then we can omit this clear action, but I think the way would get worse performance comparing with clear action.
(We do not need to clear it if we do not use early return.)

Reply:
I noticed I can omit this Clear Action by adding a guard for outside of the image size in SpdLoad().  I added the guard in my latest change. (c428856)